0.Preprocess_Data/nbconverted/0.preprocessing.py

```diff
@@ -0,0 +1,206 @@
+
+
+# In[7]:
+
+
+# Recycled code from: https://github.com/WayScience/NF1_SchwannCell_data/blob/main/5_analyze_data/notebooks/linear_model/fit_linear_model.ipynb
+cell_count_df = (
+    feature_df.groupby("Metadata_Well")["Metadata_Plate"]
+    .count()
+    .reset_index()
+    .rename(columns={"Metadata_Plate": "Metadata_number_of_singlecells"})
+)
+
+
+# In[8]:
+
+
+feature_df.head()
+
+
+# In[9]:
+
+
+# show max column in pandas df
+pd.set_option("display.max_columns", 100)
+
+
+# In[10]:
+
+
+# replace nan values with 0
+feature_df["Metadata_inducer1_concentration"] = feature_df[
+    "Metadata_inducer1_concentration"
+].fillna(0)
+feature_df["Metadata_inducer2_concentration"] = feature_df[
+    "Metadata_inducer2_concentration"
+].fillna(0)
+feature_df["Metadata_inhibitor_concentration"] = feature_df[
+    "Metadata_inhibitor_concentration"
+].fillna(0)
+
+
+# #### Combine Inducer1 and Inducer2 into one column
+
+# In[11]:
+
+
+# treatment column merge
+conditions = [
+    (feature_df["Metadata_inducer2"].isnull()),
+    feature_df["Metadata_inducer2"].notnull(),
+]
+
+results = [
+    (feature_df["Metadata_inducer1"]).astype(str),
+    (feature_df["Metadata_inducer1"] + "_" + feature_df["Metadata_inducer2"]).astype(
+        str
+    ),
+]
+feature_df["Metadata_Treatment"] = np.select(conditions, results)
+
+# dose column merge
+conditions = [
+    (feature_df["Metadata_inducer2"].isnull()),
+    feature_df["Metadata_inducer2"].notnull(),
+]
+
+results = [
+    (feature_df["Metadata_inducer1_concentration"].astype(str)).astype(str),
+    (
+        feature_df["Metadata_inducer1_concentration"].astype(str)
+        + "_"
+        + feature_df["Metadata_inducer2_concentration"].astype(str)
+    ).astype(str),
+]
+feature_df["Metadata_Dose"] = np.select(conditions, results)
+
+
+# In[12]:
+
+
+feature_df["Metadata_inducer1_concentration"] = pd.to_numeric(
+    feature_df["Metadata_inducer1_concentration"]
+)
+
+
+# ## N Beta Column condition generation
+
+# In[13]:
+
+
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "_"
+    + feature_df["Metadata_Dose"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
+
+feature_df["twob_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "_"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+    + "__"
+    + feature_df["Metadata_Dose"].astype(str)
+).astype(str)
+
+feature_df["threeb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "__"
+    + feature_df["Metadata_Dose"].astype(str)
+    + "__"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
```

Comment:
```suggestion
).astype(str)

```
Recommend adding extra space between the columns to stay consistent with the above spacing.